sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/xds/utils/RdsProtocolTransformer.java

```diff
@@ -215,4 +260,138 @@ private static XdsClusterWeight parseClusterWeight(ClusterWeight clusterWeight)
+                continue;
+            }
+            Struct headerStruct = headersMap.get("header").getStructValue();
+            XdsHeaderOption responseHeader = new XdsHeaderOption();
+            if (headersMap.get("append") != null) {
+                responseHeader.setEnabledAppend(headerStruct.getFieldsOrThrow("append").getBoolValue());
+            }
+            if (!headerStruct.containsFields("key")) {
+                continue;
+            }
+            XdsHeader xdsHeader = new XdsHeader();
+            xdsHeader.setKey(headerStruct.getFieldsOrThrow("key").getStringValue());
+            if (headerStruct.containsFields("value")) {
+                xdsHeader.setValue(headerStruct.getFieldsOrThrow("value").getStringValue());
+            }
+            responseHeader.setHeader(xdsHeader);
+            responseHeaders.add(responseHeader);
+        }
+        return responseHeaders;
+    }
+
+    private static Optional<io.sermant.core.service.xds.entity.FractionalPercent> parseRuntimeFractionalPercent(
+            Struct filterEnabledStruct) {
+        Map<String, Value> valueMap = filterEnabledStruct.getFieldsMap();
+        Value defaultValue = valueMap.get("default_value");
+        if (defaultValue == null) {
+            return Optional.empty();
+        }
+        Struct defaultValueStruct = defaultValue.getStructValue();
+        Map<String, Value> defaultValueMap = defaultValueStruct.getFieldsMap();
+        if (defaultValueMap.get("numerator") == null || defaultValueMap.get("denominator") == null) {
+            return Optional.empty();
+        }
+        io.sermant.core.service.xds.entity.FractionalPercent fractionalPercent =
+                new io.sermant.core.service.xds.entity.FractionalPercent();
+        fractionalPercent.setNumerator((int) defaultValueMap.get("numerator").getNumberValue());
+        fractionalPercent.setDenominator(FractionalPercent.DenominatorType.valueOf(defaultValueMap.get("denominator")
+                .getStringValue()).getNumber());
+        return Optional.of(fractionalPercent);
+    }
+
+    private static Optional<XdsTokenBucket> parseTokenBucket(Struct tokenBucketStruct) {
+        Map<String, Value> fieldMap = tokenBucketStruct.getFieldsMap();
+        if (fieldMap.get("tokens_per_fill") == null || fieldMap.get("fill_interval") == null
+                || fieldMap.get("max_tokens") == null) {
+            return Optional.empty();
+        }
+        double tokensPerFill = fieldMap.get("tokens_per_fill").getNumberValue();
+        long fillInterval = Duration.parse("PT" + fieldMap.get("fill_interval").getStringValue()).toMillis();
+        XdsTokenBucket xdsTokenBucket = new XdsTokenBucket();
+        xdsTokenBucket.setFillInterval(fillInterval);
+        xdsTokenBucket.setMaxTokens((int) fieldMap.get("max_tokens").getNumberValue());
+        xdsTokenBucket.setTokensPerFill((int) tokensPerFill);
+        return Optional.of(xdsTokenBucket);
+    }
```

Comment:
Too many strings needed to be managed. Besides I suggest you to give an example structure for these fileds

Reply:
done